### PR TITLE
Make compatible with TYPO3 9 and introduce invisible recaptcha

### PR DESCRIPTION
**Pull request by garbast**

### CODE REVIEW

**Comment by garbast** [comment]
Some updates to make it throw less warnings in extension scanner

**Comment by garbast** [comment]
Ping

File: Classes/Utility/LoginFailureUtility.php [hunk 1]
```diff
@@ -48,30 +29,21 @@ public static function failuresEqual($amount)
...
+                        $queryBuilder->createNamedParameter([1, 2], \Doctrine\DBAL\Connection::PARAM_STR_ARRAY)
                     ),
-                    $expression->eq(
-                        'IP',
-                        $queryBuilder->createNamedParameter(
-                            \TYPO3\CMS\Core\Utility\GeneralUtility::getIndpEnv('REMOTE_ADDR'),
-                            \PDO::PARAM_STR
-                        )
-                    ),
-                    $expression->gt('tstamp', $queryBuilder->createNamedParameter(time() - 86400, \PDO::PARAM_INT))
```
Comment (sfsmfc) [inline Classes/Utility/LoginFailureUtility.php:72]: Is there any reason for ignoring the timestamp??

File: Resources/Private/Layouts/Login.html [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+														<li class="t3js-loginprovider-switch" data-providerkey="{providerKey}"><a href="?loginProvider={providerKey}"><i class="fa fa-fw {provider.icon-class}"></i> <span><f:translate key="{provider.label}" /></span></a></li>
+													</f:if>
+												</f:if>
+											</f:for>
+										</ul>
+									</form>
+								</div>
+							</f:then>
+							<f:else>
+								<form action="../../../../../../index.php" method="post" name="loginform" id="typo3-login-form">
```
Comment (sfsmfc) [inline Resources/Private/Layouts/Login.html:72]: I think an absolute path would be the better why, or?

File: Resources/Private/Layouts8/Login.html [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+													<li class="t3js-loginprovider-switch" data-providerkey="{providerKey}"><a href="?loginProvider={providerKey}"><i class="fa fa-fw {provider.icon-class}"></i> <span><f:translate key="{provider.label}" /></span></a></li>
+												</f:if>
+											</f:if>
+										</f:for>
+									</ul>
+								</form>
+							</div>
+						</f:then>
+						<f:else>
+							<form action="../../../../../../index.php" method="post" name="loginform" id="typo3-login-form">
```
Comment (sfsmfc) [inline Resources/Private/Layouts8/Login.html:71]: I think an absolute path would be the better why, or?

Comment (sfsmfc) [reply Resources/Private/Layouts8/Login.html:71]: Is there any reason for not using the login partials from core and only adding the recaptcha?

File: ext_emconf.php [hunk 4]
```diff
@@ -3,18 +3,19 @@
...
     'priority' => 'bottom',
     'state' => 'stable',
     'clearCacheOnLoad' => 1,
     'constraints' => [
         'depends' => [
-            'typo3' => '9.4.0-9.5.99',
-            'recaptcha' => ''
+        'php' => '7.0.0-0.0.0',
+        'typo3' => '9.4.0-9.5.99',
+        'recaptcha' => ''
```
Comment (sfsmfc) [inline ext_emconf.php:18]: I think we need an under boundery for the recaptcha extension version...

**Review by sfsmfc — CHANGES_REQUESTED** [review]

File: Resources/Private/Layouts8/Login.html (hunk 3 above)
Comment (garbast) [reply Resources/Private/Layouts8/Login.html:71]: Login.html is only available as Layout and not as partial. Only the InterfaceSelector.html partials is available for login.

File: Resources/Private/Layouts/Login.html (hunk 2 above)
Comment (garbast) [reply Resources/Private/Layouts/Login.html:72]: The core does it even worth by only having "index.php"

File: Classes/Utility/LoginFailureUtility.php (hunk 1 above)
Comment (garbast) [reply Classes/Utility/LoginFailureUtility.php:72]: The timestamp is taken care by the order. It doesnt matter if one enters 5 times in a row the wrong login or leaves a month between the attempts.

File: ext_emconf.php (hunk 4 above)
Comment (garbast) [reply ext_emconf.php:18]: You could do that. I only changed that it's installable with 9.5. Since it's not possible up to now, we dont use the extension.

**Pull request closed without merging**